src/os/ui/timeline/abstracttimelinectrl.js

```diff
@@ -876,34 +876,32 @@ goog.exportProperty(
 os.ui.timeline.AbstractTimelineCtrl.prototype.openMenu = function(selector) {
   var menu = this.menus_[selector];
   if (menu) {
-    if (menu === this.zoomMenu) {
-      // clear the previous data actions
-      var dataGroup = menu.getRoot().find('Data');
-      goog.asserts.assert(!!dataGroup, 'Group "Data" should exist!');
-      dataGroup.children.length = 1;
-
-      // add data actions
-      var histData = /** @type {?Array<!os.hist.HistogramData>} */ (this['histData']);
-      if (histData) {
-        var prefix = menu === this.loadMenu ? 'load:' : 'zoom:';
-        var tip = menu === this.loadMenu ? 'Zooms to and loads ' : 'Zooms to ';
-        var sort = 100;
-
-        for (var i = 0; i < histData.length; i++) {
-          var hd = histData[i];
-          var range = hd.getRange();
-          var label = hd.getTitle();
-
-          if (hd.getVisible() && os.ui.timeline.AbstractTimelineCtrl.isSafeRange(range)) {
-            dataGroup.addChild({
-              eventType: prefix + hd.getId(),
-              label: label,
-              tooltip: tip + label,
-              icons: ['<i class="fa fa-fw fa-bars"></i>'],
-              sort: sort++,
-              handler: this.onMenuEvent.bind(this)
-            });
-          }
+    // clear the previous data actions
+    var dataGroup = menu.getRoot().find('Data');
+    goog.asserts.assert(!!dataGroup, 'Group "Data" should exist!');
```

Comment:
This throws an exception for the Load menu, which I'm guessing is why the if check was added long ago.

I would suggest adding a Data group to the load menu (so you can change the load range to match a layer/all layers), then replace the assertion with an if block (so extending classes don't hit the same exception if they add their own menu).

Reply:
@wallw-bits The data menu still has issues. The first time its opened throws an exception, then you get a duplicate menu item to load time for a layer because line 882 assumes there is an "All Data" item in the Data group and drops all children but the first.